drivers/sensor/lis2dux12/lis2dux12.c

```diff
@@ -0,0 +1,386 @@
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		lis2dux12_convert(val, data->sample_z, gain);
+		break;
+	case SENSOR_CHAN_ACCEL_XYZ:
+		lis2dux12_convert(val, data->sample_x, gain);
+		lis2dux12_convert(val + 1, data->sample_y, gain);
+		lis2dux12_convert(val + 2, data->sample_z, gain);
+		break;
+#if defined(CONFIG_LIS2DUX12_ENABLE_TEMP)
+	case SENSOR_CHAN_DIE_TEMP:
+		sensor_value_from_double(val, data->sample_temp);
+		break;
+#endif
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int lis2dux12_channel_get(const struct device *dev, enum sensor_channel chan,
+				 struct sensor_value *val)
+{
+	struct lis2dux12_data *data = dev->data;
+
+	return lis2dux12_get_channel(chan, val, data, data->gain);
+}
+
+static const struct sensor_driver_api lis2dux12_driver_api = {
+	.attr_set = lis2dux12_attr_set,
+#if defined(CONFIG_LIS2DUX12_TRIGGER)
+	.trigger_set = lis2dux12_trigger_set,
+#endif
+	.sample_fetch = lis2dux12_sample_fetch,
+	.channel_get = lis2dux12_channel_get,
+};
+
+static int lis2dux12_init(const struct device *dev)
+{
+	const struct lis2dux12_config *const cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	uint8_t chip_id;
+	int ret;
+
+	lis2dux12_exit_deep_power_down(ctx);
+	k_busy_wait(25000); /* 25ms */
+
+	/* check chip ID */
+	ret = lis2dux12_device_id_get(ctx, &chip_id);
+	if (ret < 0) {
+		LOG_ERR("%s: Not able to read dev id", dev->name);
+		return ret;
+	}
+
+	if (chip_id != LIS2DUX12_ID) {
+		LOG_ERR("%s: Invalid chip ID 0x%02x", dev->name, chip_id);
+		return -EINVAL;
+	}
+
+	/* reset device */
+	ret = lis2dux12_init_set(ctx, LIS2DUX12_RESET);
+	if (ret < 0) {
+		return ret;
+	}
+
+	k_busy_wait(100);
+
+	LOG_WRN("%s: chip id 0x%x", dev->name, chip_id);
+
+	/* Set bdu and if_inc recommended for driver usage */
+	lis2dux12_init_set(ctx, LIS2DUX12_SENSOR_ONLY_ON);
+
+	static lis2dux12_fifo_mode_t fifo_mode;
+	/* Set FIFO watermark to 32 sample(s) */
+	fifo_mode.store = LIS2DUX12_FIFO_1X;
+	fifo_mode.xl_only = 0;
+	fifo_mode.watermark = 33;                    /* 32 samples + timestamp */
+	fifo_mode.operation = LIS2DUX12_STREAM_MODE; /* old data is automatically overwritten */
+	fifo_mode.batch.dec_ts = LIS2DUX12_DEC_TS_OFF;
+	fifo_mode.batch.bdr_xl = LIS2DUX12_BDR_XL_ODR;
+	lis2dux12_fifo_mode_set(ctx, fifo_mode);
+
+	lis2dux12_timestamp_set(ctx, PROPERTY_ENABLE);
+
+#ifdef CONFIG_LIS2DUX12_TRIGGER
+	if (cfg->trig_enabled) {
+		ret = lis2dux12_trigger_init(dev);
+		if (ret < 0) {
+			LOG_ERR("%s: Failed to initialize triggers", dev->name);
+			return ret;
+		}
+	}
+#endif
+
+	/* set sensor default pm and odr */
+	LOG_WRN("%s: pm: %d, odr: %d", dev->name, cfg->pm, cfg->odr);
+	lis2dux12_md_t mode = {
+		.odr = cfg->odr,
+		.fs = cfg->range,
+	};
+	ret = lis2dux12_mode_set(ctx, &mode);
+	if (ret < 0) {
+		LOG_ERR("%s: odr init error (12.5 Hz)", dev->name);
+		return ret;
+	}
+
+	/* set sensor default scale (used to convert sample values) */
+	LOG_WRN("%s: range is %d", dev->name, cfg->range);
```

Comment:
Why LOG_WRN? They are just information. So, I would put LOG_INF. Change all.

Reply:
If there are errors (i.e. return -1) put LOG_ERR. I saw a LOG_WRN somewhere that should be LOG_ERR.

Reply:
Good catch. I changed the `WRN` to `INF` and combed the rest of the file for places where it should be `ERR` instead of `WRN`.